content/en/platform/corda/5.0-dev-preview-1/tutorials/run-demo-cordapp.md

```diff
@@ -79,54 +79,45 @@ The project containing the sample CorDapp opens.
+This command builds the CorDapp package (`.cpk`) files. Most CorDapps will have two files: a `contracts` file and a `workflows` file. The command also builds the CorDapp package bundle (`.cpb`).
 
-4. Run a Gradle command to build the CorDapp and generate the `.cpk` files:
+4. Deploy the network using `corda-cli` and `docker-compose`:
+  `corda-cli network deploy -n solar-system | docker-compose -f - up`
 
-  `gradlew build`
-
-5. Use the `cordapp-builder` CLI utility to bundle the `.cpk` files into an easy-to-use `.cpb` file:
-
-  `cordapp-builder create --cpk contracts\build\libs\corda5-solar-system-contracts-cordapp.cpk --cpk workflows\build\libs\corda5-solar-system-workflows-cordapp.cpk -o corda5-hello-solarsystem.cpb`
-
-This command specifies the `.cpk` files of your CorDapp. Most CorDapps will have two files: a `contracts` file and a `workflows` file.
-
-This command also specifies the output file. In this case, the output file is `corda5-hello-solarsystem.cpb`.
-
-  If the command is successful, there is no output.
-
-  If you have an error in your command, the segment with the error is returned in red.
-
-6. Deploy the network using `docker-compose`:
-  `docker-compose -f docker-compose.yaml up`
+{{< note >}}
+The safest and preferred method for seeing the contents of the network is by piping to Docker.
 
-  This command references the name of the `docker-compose` file generated in step 2.
+While this is *not* the recommended approach, you can also output the contents to a file if you want to see what is happening. Use this command to do so: `corda-cli network deploy -n solar-system -f solar-system.yaml > solar-system-compose.yaml`.
+{{< /note >}}
 
-7. Check the CorDapp's status with Corda CLI:
+5. Check the CorDapp's status using [Corda CLI](../../../../../en/platform/corda/5.0-dev-preview-1/corda-cli/overview.md):
   `corda-cli network status -n solar-system`
 
   You'll be able to see the status of the node. The nodes are up and running when their status is `Ready`.
 
   {{< note >}}
   Take note of the `HTTP RPC port` for each node. You will use these later when you [test the CorDapp using Swagger UI](#test-the-sample-cordapp-using-swagger-ui) or [Corda Node CLI](#test-the-sample-cordapp-using-corda-node-cli).
   {{< /note >}}
-8. Install the application on the network using Corda CLI.
+
+6. Alternatively, wait for the network to be ready with:
+`corda-cli network wait -n solar-system`
+
+7. Install the application on the network using Corda CLI.
 
 In Corda 4, this process was much more involved. Now you can install the application on the network with a single command:
 
   `corda-cli package install -n solar-system cordaSolarSystem.cpb`
 
-  In this command, you must specify the network and the `.cpb` file.
+  In this command, you must specify the network and the `.cpb` file. Depending on the Gradle setupm the `.cpb` will be in one of the build folders and the name may be different.
```

Comment:
This seems to be not the case. The gradle build of the demo project (current version `commit 404cdd743a063e04b3dfe62ded943bac589f0624, Wed Sep 29 16:11:59 2021 +0100`) does not produce the `.cpb` file. 

It needs to be created by the original step 5 (line 95):
``cordapp-builder create --cpk contracts\build\libs\corda5-solar-system-contracts-cordapp.cpk --cpk workflows\build\libs\corda5-solar-system-workflows-cordapp.cpk -o corda5-hello-solarsystem.cpb``.


The steps which worked for me were documented with #277 (which was closed in favour of this PR).

Reply:
The cpb pluggin only needs to be applied to the "main" module. In this case, the workflow module. Because that declares a dependency on the contracts module, both CPKs will be packaged up in the resulting cpb.
The cpb plugin only needs to be in the workflow gradle file (`workflows/build.gradle`):
```
plugins {
    id 'net.corda.plugins.cordapp-cpb'
}
```
The contract module only needs the cpk plugin.

Optionally you can then change the default name and location of the built cpb like so (for example):
```
cpb {
    archiveBaseName = 'solar-system'
    destinationDir = file("$rootDir/build/")
}
```

I created a PR to illustrate this: https://github.com/corda/corda5-solarsystem-contracts-demo/pull/2

Reply:
Thanks for your help on this one, @pharsfalvi !